From e4c388d2a0d4d35602c04b82b4451eb978a59564 Mon Sep 17 00:00:00 2001
From: Cerbero Build System <cerbero@gstreamer.freedesktop.org>
Date: Fri, 11 Mar 2022 02:14:12 +0000
Subject: [PATCH] qtdemux: Raise max atom size

---
 gst/isomp4/qtdemux.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gst/isomp4/qtdemux.c b/gst/isomp4/qtdemux.c
index b74f4ef..41d4800 100644
--- a/gst/isomp4/qtdemux.c
+++ b/gst/isomp4/qtdemux.c
@@ -83,7 +83,7 @@
 #endif
 
 /* max. size considered 'sane' for non-mdat atoms */
-#define QTDEMUX_MAX_ATOM_SIZE (32*1024*1024)
+#define QTDEMUX_MAX_ATOM_SIZE (100*1024*1024)
 
 /* if the sample index is larger than this, something is likely wrong */
 #define QTDEMUX_MAX_SAMPLE_INDEX_SIZE (200*1024*1024)
-- 
2.30.2

