From ddc87f3a38c990826b8be3a28cf395139244a91a Mon Sep 17 00:00:00 2001
From: Jake Barnes <me@jakebarn.es>
Date: Thu, 6 Aug 2020 16:46:00 +1000
Subject: [PATCH] Add missing GST_STATIC_COMPILATION and ORC_STATIC_COMPILATION

---
 meson.build | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/meson.build b/meson.build
index 5ee26c8..a0f49b6 100644
--- a/meson.build
+++ b/meson.build
@@ -317,6 +317,10 @@ glib_deps = [dependency('glib-2.0', version : glib_req, fallback: ['glib', 'libg
 gio_dep = dependency('gio-2.0', fallback: ['glib', 'libgio_dep'])
 
 gst_plugins_good_args = ['-DHAVE_CONFIG_H']
+if get_option('default_library') == 'static'
+  gst_plugins_good_args += ['-DGST_STATIC_COMPILATION', '-DORC_STATIC_COMPILATION']
+endif
+
 configinc = include_directories('.')
 libsinc = include_directories('gst-libs')
 
-- 
2.26.2.windows.1

