From 637a97d8e1838b58eb53489179edcd517bcfd1ae Mon Sep 17 00:00:00 2001
From: Mark Fernie <39318992+drmarkfernie@users.noreply.github.com>
Date: Mon, 11 Nov 2019 15:50:09 +1100
Subject: [PATCH] increase rank of vp8dec

---
 ext/vpx/plugin.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/vpx/plugin.c b/ext/vpx/plugin.c
index 4f04efa..a49c253 100644
--- a/ext/vpx/plugin.c
+++ b/ext/vpx/plugin.c
@@ -34,7 +34,7 @@ static gboolean
 plugin_init (GstPlugin * plugin)
 {
 #ifdef HAVE_VP8_DECODER
-  gst_element_register (plugin, "vp8dec", GST_RANK_PRIMARY,
+  gst_element_register (plugin, "vp8dec", GST_RANK_PRIMARY + 100,
       gst_vp8_dec_get_type ());
 #endif
 
-- 
2.17.1

