From cf0278f5014b010ab1a9d8ed5aa5d9d08e17008d Mon Sep 17 00:00:00 2001
From: Cerbero Build System <cerbero@gstreamer.freedesktop.org>
Date: Fri, 11 Mar 2022 03:37:54 +0000
Subject: [PATCH] soup: Just set User-Agent on the session

---
 ext/soup/gstsouphttpsrc.c | 22 +---------------------
 1 file changed, 1 insertion(+), 21 deletions(-)

diff --git a/ext/soup/gstsouphttpsrc.c b/ext/soup/gstsouphttpsrc.c
index 003a113..57a7b61 100644
--- a/ext/soup/gstsouphttpsrc.c
+++ b/ext/soup/gstsouphttpsrc.c
@@ -997,7 +997,7 @@ thread_func (gpointer user_data)
    * to be able to have the same session with different User-Agents per
    * source */
   session->session =
-      _soup_session_new_with_options ("user-agent", NULL,
+      _soup_session_new_with_options ("user-agent", src->user_agent,
       "timeout", src->timeout, "tls-interaction", src->tls_interaction,
       /* Unset the limit the number of maximum allowed connections */
       "max-conns", src->session_is_shared ? G_MAXINT : 10,
@@ -1727,26 +1727,6 @@ gst_soup_http_src_build_message (GstSoupHTTPSrc * src, const gchar * method)
 
   request_headers = _soup_message_get_request_headers (src->msg);
 
-  /* Duplicating the defaults of libsoup here. We don't want to set a
-   * User-Agent in the session as each source might have its own User-Agent
-   * set */
-  if (!src->user_agent || !*src->user_agent) {
-    gchar *user_agent =
-        g_strdup_printf ("libsoup/%u.%u.%u", _soup_get_major_version (),
-        _soup_get_minor_version (), _soup_get_micro_version ());
-    _soup_message_headers_append (request_headers, "User-Agent", user_agent);
-    g_free (user_agent);
-  } else if (g_str_has_suffix (src->user_agent, " ")) {
-    gchar *user_agent = g_strdup_printf ("%slibsoup/%u.%u.%u", src->user_agent,
-        _soup_get_major_version (),
-        _soup_get_minor_version (), _soup_get_micro_version ());
-    _soup_message_headers_append (request_headers, "User-Agent", user_agent);
-    g_free (user_agent);
-  } else {
-    _soup_message_headers_append (request_headers, "User-Agent",
-        src->user_agent);
-  }
-
   if (!src->keep_alive) {
     _soup_message_headers_append (request_headers, "Connection", "close");
   }
-- 
2.30.2