From 87b6631163a23951b93851dd2a0adc54148fa848 Mon Sep 17 00:00:00 2001
From: Tristan Penman <tristan@tristanpenman.com>
Date: Mon, 2 Nov 2020 16:02:18 +1100
Subject: [PATCH] Set proxy username and password from env var

---
 ext/soup/gstsouphttpsrc.c | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/ext/soup/gstsouphttpsrc.c b/ext/soup/gstsouphttpsrc.c
index e75a0aad7..c612d929d 100644
--- a/ext/soup/gstsouphttpsrc.c
+++ b/ext/soup/gstsouphttpsrc.c
@@ -2119,6 +2119,26 @@ gst_soup_http_src_set_proxy (GstSoupHTTPSrc * src, const gchar * uri)
     g_free (new_uri);
   }
 
+  if (src->proxy == NULL) {
+    return FALSE;
+  }
+
+  if (soup_uri_get_user (src->proxy) != NULL) {
+    if (src->proxy_id) {
+      g_free (src->proxy_id);
+      src->proxy_id = NULL;
+    }
+    src->proxy_id = g_strdup (soup_uri_get_user (src->proxy));
+  }
+
+  if (soup_uri_get_password (src->proxy) != NULL) {
+    if (src->proxy_pw) {
+      g_free (src->proxy_pw);
+      src->proxy_pw = NULL;
+    }
+    src->proxy_pw = g_strdup (soup_uri_get_password (src->proxy));
+  }
+
   return (src->proxy != NULL);
 }
 
-- 
2.11.0

