From 5fdc3eec417e6ce91dbb081d886ce4fcce716c43 Mon Sep 17 00:00:00 2001
From: Andrew Pritchard <andrew@vivi.io>
Date: Mon, 28 Mar 2022 14:02:27 +1100
Subject: [PATCH] Proxy stuff

---
 ext/soup/gstsouphttpsrc.c | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/ext/soup/gstsouphttpsrc.c b/ext/soup/gstsouphttpsrc.c
index 7204c1bf29..0ac9c949a1 100644
--- a/ext/soup/gstsouphttpsrc.c
+++ b/ext/soup/gstsouphttpsrc.c
@@ -72,6 +72,8 @@
 #include "config.h"
 #endif
 
+#include <libsoup/soup.h>
+
 #include <string.h>
 #ifdef HAVE_STDLIB_H
 #include <stdlib.h>             /* atoi() */
@@ -2568,6 +2570,26 @@ gst_soup_http_src_set_proxy (GstSoupHTTPSrc * src, const gchar * uri)
     g_free (new_uri);
   }
 
+  if (!src->proxy || !src->proxy->soup_uri) {
+    return FALSE;
+  }
+
+  if (soup_uri_get_user (src->proxy->soup_uri) != NULL) {
+    if (src->proxy_id) {
+      g_free (src->proxy_id);
+      src->proxy_id = NULL;
+    }
+    src->proxy_id = g_strdup (soup_uri_get_user (src->proxy->soup_uri));
+  }
+
+  if (soup_uri_get_password (src->proxy->soup_uri) != NULL) {
+    if (src->proxy_pw) {
+      g_free (src->proxy_pw);
+      src->proxy_pw = NULL;
+    }
+    src->proxy_pw = g_strdup (soup_uri_get_password (src->proxy->soup_uri));
+  }
+
   return (src->proxy != NULL);
 }
 
-- 
2.32.0 (Apple Git-132)


