From c5580f76515ba6d9a3c0b481c7853e5b5bbf7a9e Mon Sep 17 00:00:00 2001
From: Jake Barnes <me@jakebarn.es>
Date: Tue, 23 Jun 2020 17:57:51 +1000
Subject: [PATCH] mediafoundation: Make device provider actually work

---
 sys/mediafoundation/gstmfdevice.c | 22 ++++++++++++++++------
 1 file changed, 16 insertions(+), 6 deletions(-)

diff --git a/sys/mediafoundation/gstmfdevice.c b/sys/mediafoundation/gstmfdevice.c
index f11b7a8a8..b3c5aec41 100644
--- a/sys/mediafoundation/gstmfdevice.c
+++ b/sys/mediafoundation/gstmfdevice.c
@@ -169,9 +169,16 @@ gst_mf_device_provider_probe (GstDeviceProvider * provider)
 {
   GstMFDeviceProvider *self = GST_MF_DEVICE_PROVIDER (provider);
   GList *list = NULL;
-  gint i;
+  GList *activate_list = NULL;
+  GList *iter = NULL;
+
+  if (!gst_mf_source_enum_device_activate (GST_MF_SOURCE_TYPE_VIDEO, &activate_list)) {
+    GST_WARNING_OBJECT (self, "No available video capture device");
+  }
+
+  for (iter = activate_list; iter; iter = g_list_next (iter)) {
+    GstMFDeviceActivate *activate = (GstMFDeviceActivate *) iter->data;
 
-  for (i = 0;; i++) {
     GstMFSourceObject *obj = NULL;
     GstDevice *device;
     GstStructure *props = NULL;
@@ -179,20 +186,19 @@ gst_mf_device_provider_probe (GstDeviceProvider * provider)
     gchar *device_name = NULL;
     gchar *device_path = NULL;
 
-
 #if HAVE_CAPTURE_ENGINE
-    obj = gst_mf_capture_engine_new (GST_MF_SOURCE_TYPE_VIDEO, i, NULL, NULL);
+    obj = gst_mf_capture_engine_new (GST_MF_SOURCE_TYPE_VIDEO, activate->index, activate->name, activate->path);
 #endif
 
     if (!obj)
-      obj = gst_mf_source_reader_new (GST_MF_SOURCE_TYPE_VIDEO, i, NULL, NULL);
+      obj = gst_mf_source_reader_new (GST_MF_SOURCE_TYPE_VIDEO, activate->index, activate->name, activate->path);
 
     if (!obj)
       break;
 
     caps = gst_mf_source_object_get_caps (obj);
     if (!caps) {
-      GST_WARNING_OBJECT (self, "Empty caps for device index %d", i);
+      GST_WARNING_OBJECT (self, "Empty caps for device index %d", activate->index);
       goto next;
     }
 
@@ -230,5 +236,9 @@ gst_mf_device_provider_probe (GstDeviceProvider * provider)
     gst_object_unref (obj);
   }
 
+  if (activate_list)
+    g_list_free_full(activate_list,
+        (GDestroyNotify) gst_mf_device_activate_free);
+
   return list;
 }
-- 
2.26.2.windows.1

