From ee2930b3acb6ee849872709d8578204f536236c5 Mon Sep 17 00:00:00 2001
From: Mark Fernie <39318992+drmarkfernie@users.noreply.github.com>
Date: Tue, 1 Oct 2019 14:13:11 +1000
Subject: [PATCH] gstvolume: remove GAP FLAG as it causes SEGV in
 audio_base_sink

---
 gst/volume/gstvolume.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/gst/volume/gstvolume.c b/gst/volume/gstvolume.c
index fc10191..e875ce8 100644
--- a/gst/volume/gstvolume.c
+++ b/gst/volume/gstvolume.c
@@ -795,7 +795,6 @@ volume_transform_ip (GstBaseTransform * base, GstBuffer * outbuf)
 
   if (self->current_volume == 0.0 || self->current_mute) {
     orc_memset (map.data, 0, map.size);
-    GST_BUFFER_FLAG_SET (outbuf, GST_BUFFER_FLAG_GAP);
   } else if (self->current_volume != 1.0) {
     self->process (self, map.data, map.size);
   }
-- 
2.17.1

