From d9db3731e75b234a2d707e5542ab5cd9eb8ef4bf Mon Sep 17 00:00:00 2001
From: Jake Barnes <me@jakebarn.es>
Date: Thu, 16 Apr 2020 10:05:18 +1000
Subject: [PATCH] Bump avdec_aac rank above primary

---
 ext/libav/gstavauddec.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/ext/libav/gstavauddec.c b/ext/libav/gstavauddec.c
index d518b70..83b21ef 100644
--- a/ext/libav/gstavauddec.c
+++ b/ext/libav/gstavauddec.c
@@ -839,10 +839,12 @@ gst_ffmpegauddec_register (GstPlugin * plugin)
      * msmpeg4v3 same, as it outperforms divxdec for divx3 playback.
      * VC1/WMV3 are not working and thus unpreferred for now. */
     switch (in_plugin->id) {
+      case AV_CODEC_ID_AAC:
+        rank = GST_RANK_PRIMARY + 100;
+        break;
       case AV_CODEC_ID_RA_144:
       case AV_CODEC_ID_RA_288:
       case AV_CODEC_ID_COOK:
-      case AV_CODEC_ID_AAC:
         rank = GST_RANK_PRIMARY;
         break;
         /* SIPR: decoder should have a higher rank than realaudiodec.
-- 
2.26.1

